routing/turns_tts_text.cpp

```diff
@@ -47,10 +49,243 @@ void GetTtsText::ForTestingSetLocaleWithJson(std::string const & jsonBuffer, std
+ * @return 1 = front = -re, 2 = back = -ra
+ */
+uint8_t CategorizeHungarianAcronymsAndNumbers(std::string const & myString)
+{
+  constexpr std::array<std::string_view, 14> backNames = {
+      "A", // a
+      "Á", // á
+      "H", // há
+      "I", // i
+      "Í", // í
+      "K", // ká
+      "O", // o
+      "Ó", // ó
+      "U", // u
+      "Ű", // ú
+      "0", // nulla or zéró
+      "3", // három
+      "6", // hat
+      "8", // nyolc
+  };
+
+  constexpr std::array<std::string_view, 31> frontNames = {
+      // all other letters besides H and K
+      "B", "C", "D", "E", "É", "F", "G", "J", "L", "M", "N", "Ö", "Ő", "P", "Q", "R", "S", "T", "Ú", "Ü", "V", "W", "X", "Y", "Z",
+      "1", // egy
+      "2", // kettő
+      "4", // négy
+      "5", // öt
+      "7", // hét
+      "9", // kilenc
+  };
+
+  constexpr std::array<std::string_view, 5> specialCaseFront = {
+      "10", // tíz special case front
+      "40", // negyven front
+      "50", // ötven front
+      "70", // hetven front
+      "90", // kilencven front
+  };
+
+  constexpr std::array<std::string_view, 4> specialCaseBack = {
+      "20", // húsz back
+      "30", // harminc back
+      "60", // hatvan back
+      "80", // nyolcvan back
+  }; //TODO: consider regex
+
+  //'100', // száz back
+
+  for (int i = myString.size()-1; i >= 0; i--) {
+    strings::UniString original = strings::MakeUniString(myString);
+    // special is 2 char, so check last 2
+    std::string oneBuf = strings::ToUtf8(strings::Substring(original, i, 1));
+    std::string twoBuf = strings::ToUtf8(strings::Substring(original, i-1, 2));
+    std::string threeBuf = strings::ToUtf8(strings::Substring(original, i-2, 3));
+    if (FindInStrArray(specialCaseFront, twoBuf) != -1)
+      return 1;
+    if (FindInStrArray(specialCaseBack, twoBuf) != -1)
+      return 2;
+    if (threeBuf == "100")
+      return 2;
+
+    if (FindInStrArray(frontNames, oneBuf) != -1)
+      return 1;
+    if (FindInStrArray(backNames, oneBuf) != -1)
+      return 2;
+    if (oneBuf == " ") // if we've somehow hit a space, just say it's back
+      return 2;
+  }
+
+  LOG(LWARNING, ("Unable to find Hungarian front/back for", myString));
+  return 2;
+}
+
+/**
+ * @brief Decides if a string (possibly Hungarian) has a "front" or "back" ending.
+ *
+ * Much like the Acronym/Number algorithm above, we start from the back of the word and
+ * keep trying to match a front or back vowel until we find one. Indeterminate vowels are
+ * "back" by default but only if we find nothing else. And if we truly find nothing, it
+ * may be an acronym after all. (The word "acerbic" has a different ending sound than ABC.)
+ *
+ * @arg std::string const & myString - words, numbers or acronyms to be spoken in Hungarian
+ * @return 1 = front = -re, 2 = back = -ra
+ */
+uint8_t CategorizeHungarianLastWordVowels(std::string const & myString)
+{
+  constexpr std::array<std::string_view, 6> front = {"e","é","ö","ő","ü","ű"};
```

Comment:
Can all characters be defined as `U'e'` char32_t code points, and all work/transformation/comparison be done using UniString, without unnecessary conversions?

`std::u32string_view` can also be used if necessary.